src/arrays/arraymath.jl

```diff
@@ -61,6 +61,22 @@ end
 
 *(dm1::DualMatrix, dm2::DualMatrix) = (dm1.qarr*dm2.qarr)'
 
+function +{B<:OrthonormalBasis,N}(qarr1::Union(QuArray{B,N},CTranspose{B,N}), qarr2::Union(QuArray{B,N},CTranspose{B,N}))
+    if bases(qarr1) == bases(qarr2)
+        return QuArray(coeffs(qarr1)+coeffs(qarr2), bases(qarr1))
+    else
+        error("Bases not compatible")
+  end
+end
+
+function -{B<:OrthonormalBasis,N}(qarr1::Union(QuArray{B,N},CTranspose{B,N}), qarr2::Union(QuArray{B,N},CTranspose{B,N}))
+    if bases(qarr1) == bases(qarr2)
+        return QuArray(coeffs(qarr1)-coeffs(qarr2), bases(qarr1))
+    else
+        error("Bases not compatible")
+  end
+end
+
```

Comment:
I know that we already have this pattern in the code, but maybe we should be annotating arguments as `AbstractQuArray{B,N}` instead of `Union(QuArray{B,N},CTranspose{B,N})` for functions like these. It's both less wordy and more generic. 

Thoughts, @acroy?